### PR TITLE
Make multiprocessing.Queue a subclass of queue.Queue

### PR DESCRIPTION
**Pull request by bcl**
Also change multiprocessing.Queue's put and get timeout arguments to
allow None.

This fixes a problem with logging.handlers.QueueHandler and
QueueListener not accepting a multiprocessing.Queue as the queue
argument.

### CODE REVIEW

File: stdlib/3/multiprocessing/__init__.pyi [hunk 1]
```diff
@@ -93,10 +94,10 @@ class Process():
     def is_alive(self) -> bool: ...
     def join(self, timeout: Optional[float] = ...) -> None: ...
 
-class Queue():
+class Queue(queue.Queue):
```
Comment (gvanrossum) [inline stdlib/3/multiprocessing/__init__.pyi]: So this implicitly makes the base class `queue.Queue[Any]`. I wonder if it wouldn't be better to make this `Queue` class generic as well, so one can instantiate a `mp.Queue[int]`, say.

Comment (bcl) [inline stdlib/3/multiprocessing/__init__.pyi]: I'm not that familiar with the mypy types yet, so I'm not sure if I'm doing this correctly, but changing it to a `Generic[_T]` like queue.Queue gives me `error: Need type annotation for variable` when I do `q = mp.Queue()`

**Comment by gvanrossum** [comment]
Hm, that's also the case when instantiating `queue.Queue()` so maybe this is a feature? I'm reluctant to keep `mp.Queue` non-generic. Take the [example from the docs](https://docs.python.org/3.6/library/multiprocessing.html#exchanging-objects-between-processes), the type of `q` should probably be declared as `Queue[List[Any]]`, so that the type-checker will know that `q.get()` returns a `List`.

**Comment by bcl** [comment]
Ah! Right, that makes sense.

**Comment by bcl** [comment]
I'm not sure what's up with that travis failure. When I run the tests here they all pass:

    SUMMARY  all 151 tasks and 1451 tests passed                                    
    *** OK ***
    total runtime: 166.2158714469988 sec
    mypy tests succeeded

**Comment by gvanrossum** [comment]
I've restarted the test job. I believe it's an intermittent Travis failure: https://github.com/python/mypy/issues/3543.

File: stdlib/3/multiprocessing/__init__.pyi [hunk 2]
```diff
@@ -93,10 +96,10 @@ class Process():
     def is_alive(self) -> bool: ...
     def join(self, timeout: Optional[float] = ...) -> None: ...
 
-class Queue():
+class Queue(queue.Queue[_T]):
     def __init__(self, maxsize: int = ...) -> None: ...
-    def get(self, block: bool = ..., timeout: float = ...) -> Any: ...
-    def put(self, item: Any, block: bool = ..., timeout: float = ...) -> None: ...
+    def get(self, block: bool = ..., timeout: Optional[float] = ...) -> Any: ...
+    def put(self, item: Any, block: bool = ..., timeout: Optional[float] = ...) -> None: ...
```
Comment (gvanrossum) [inline stdlib/3/multiprocessing/__init__.pyi]: The two `Any`s in `get()` and `put()` should be replaced with `_T`, to match the same methods in `queue.Queue`.

**Comment by bcl** [comment]
Oops, sorry about that. Also changed Any in put_nowait and get_nowait to _T

**Comment by gvanrossum** [comment]
Thanks! I am still a little worried that this may break code that's using this without an annotation; we'll see how it goes in practice.

**Comment by gvanrossum** [comment]
Oh, PS. Next time please don't squash your local commits -- this makes it harder to track a code review, and it's not needed, since we already squash when we merge into master.

**Comment by bcl** [comment]
Will do. Thanks!

**Pull request merged**